src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -991,6 +991,20 @@ public virtual void Save(string filePath, bool isBackup = false, EngineControlle
             try
             {
                 // Stage 1: Serialize the workspace.
+                string fileName = string.Empty;
+                try
+                {
+                    fileName = Path.GetFileName(filePath);
+                    string extension = Path.GetExtension(filePath);
+                    if (extension == ".dyn" || extension == ".dyf")
+                    {
+                        fileName = Path.GetFileNameWithoutExtension(filePath);
+                    }
+                }
+                catch (ArgumentException)
```

Comment:
@ColinDayOrg  Can you remind me what is `ArgumentException` catch here for?

Reply:
I don't remember exactly off the top of my head which one, but one of the Path methods being called throws an exception if the arguments are not as expected. It was something like GetExtension throws if there is no extension or something like that. Bad API design causing exceptions to be used for logical control. I can look deeper and try to remember if you think it is important.

Reply:
@ColinDayOrg I see, I just want to confirm since we are catching but not doing anything there, so I was curious if we can just throw in this case.

Reply:
If I remember correctly, in this case throwing was just a logical indication that the filename should not be set, as opposed to actually rethrowing which would cause the save to fail.